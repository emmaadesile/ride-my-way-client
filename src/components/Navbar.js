import React, { Component } from 'react';
import { Link } from '@reach/router';

class Navbar extends Component {
  state = {
    isLoggeIn: false
  }
  render() {

    return (
      <div>
        {
          this.state.isLoggeIn
          ? <ul>
              {/* <li><Link to='/'><img src={logo} alt='logo' /></Link></li> */}
              <li><Link to='/about'>About</Link></li>
              <li><Link to='/auth/login'>Logout</Link></li>
              <li><Link to='/rides'>Rides</Link></li>
              <li><Link to='/ride-requests'>Requests</Link></li>
              <li><Link to='/ride-history'>Ride History</Link></li>
              <li><Link to='/user-profile'>Your Profile</Link></li>
            </ul>
          : <ul>
              {/* <li><Link to='/'><img src={logo} alt='logo' /></Link></li> */}
              <li><Link to='/about'>About</Link></li>
              <li><Link to='/auth/login'>Login</Link></li>
              <li><Link to='/auth/signup'>Signup</Link></li>
            </ul>
        }
      </div>
    )
  }
}

export default Navbar;