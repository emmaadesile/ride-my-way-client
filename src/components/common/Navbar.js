import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import HamburgerMenu from 'react-hamburger-menu';
import '../../assets/styles/style.css';
import '../../assets/images/hamburger.svg';

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    }
  }

  handleClick() {
    this.setState({
      open: !this.state.open
    })
  }
  render() {
    return (
      <div className="navbar box-shadow-none">
        <ul>
          <li className="logo">
            <Link to="/">
              <img src={require('../../assets/images/ride-logo.png')} alt="logo" />
            </Link>
          </li>
  
          <section className="div_navbar_items">
            <li className="navbar_items"> <Link to="/signup"> Sign Up </Link> </li>
            <li className=" navbar_items ">
              <Link to="/login"> Sign In </Link>
            </li>
          </section>
  
          <li className="icon">
            <HamburgerMenu
              isOpen={this.state.open}
              menuClicked={this.handleClick.bind(this)}
              width={18}
              height={15}
              strokeWidth={2}
              rotate={0}
              color='white'
              borderRadius={0}
              animationDuration={0.5}
            />
          </li>
        </ul>
      </div>
    )
  }
}

export default Navbar;