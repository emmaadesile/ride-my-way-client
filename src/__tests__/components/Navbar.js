import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import toJson from 'enzyme-to-json';
import Navbar from "../../components/common/Navbar";

const mockStore = configureMockStore();
const store = mockStore();

describe("Navbar Component", () => {
  test("renders the Navbar commpinent", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Navbar />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Navbar />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
