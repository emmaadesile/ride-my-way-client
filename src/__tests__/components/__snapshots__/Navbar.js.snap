// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `<Connect(Navbar) />`;
